src/test/kotlin/com/igorwojda/string/capitalizesentence/challenge.kt

```diff
@@ -4,7 +4,7 @@ import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.Test
 
 private fun capitalizeSentence(str: String): String {
-    TODO("not implemented")
+    return str.split("\\s".toRegex()).reduce { acc, s ->acc +' '+s.capitalize() }.toString().capitalize()
```

Comment:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/join-to-string.html is a powerful tool